Changelog for v2.172.0 (#3769)
vertexproject/synapse: CHANGELOG.rst

Synapse Changelog
*****************

v2.172.0 - 2024-06-22
=====================

Model Changes
-------------
- Updates to the ``biz``, ``econ``, ``inet``, ``meta``, ``ou`` ``risk``,
 and ``transit`` models.

>   (`#3561 <https://github.cosem/vertexproject/synapse/pull/3561>`_)

Reviewer: ```suggestion
  (`#3561 <https://github.com/vertexproject/synapse/pull/3561>`_)
```

  (`#3756 <https://github.com/vertexproject/synapse/pull/3756>`_)

  **New Interfaces**

  ``inet:service:base``
    Properties common to most forms within a service platform.

  ``inet:service:object``
    Properties common to objects within a service platform. This inherits
    from the ``inet:service:base`` interface.

>   **New Forms**

Reviewer: Feel free to say no:

Can we sort this section lexicographically by form name? It would help for visually scanning for a specific form. Probably the "New Properties" section below too.

  ``inet:service:access``
    Represents a user access request to a service resource.

  ``inet:service:account``

>    An account within a service platform. Accounts may be instance specific.

Reviewer 2: Missing one space at the start

    ``type``
      The type of network.

  ``meta:note``
    The form had the following property added to it:

    ``replyto``
      The note is a reply to the specified note.

  ``ou:campaign``
    The form had the following property added to it:

    ``ext:id``
      An external identifier for the campaign.

  ``ou:org``
    The form had the following property added to it:

    ``ext:id``
      An external identifier for the organization.

  ``ou:technique``
    The form had the following property added to it:

    ``ext:id``
      An external identifier for the technique.

  ``risk:extortion``
    The form had the following properties added to it:

    ``paid:price``
      The total price paid by the target of the extortion.

    ``payments``
      Payments made from the target to the attacker.

  ``risk:leak``
    The form had the following properties added to it:

    ``size:count``
      The number of files included in the leaked data.

    ``size:percent``
      The total percent of the data leaked.

  ``risk:threat``
    The form had the following property added to it:

    ``ext:id``
      An external identifier for the threat.

  **Updated Types**

  ``inet:web:hashtag``

>     Loosen the regex for hashtag validation.

Reviewer: Do you want mention it supporting unicode now?

Author: It's always supported unicode word characters. We only added the middle dot (U+00B7) as an allowed character AFTER the first unicode word character in the hashtag. We could be clear about that.

  ``transport:air:flightnum``
    Loosen the regex for flight number validation.

  **Updated Forms**

  ``it:host``
    The form now inherits from the ``inet:service:object`` interface.

Features and Enhancements
-------------------------

> - Update the permission checking for View mergeing ( ``view.merge`` ) to
>   optimize the permission checking based user permissions and layer index

Reviewer: ```suggestion
- Update the permission checking for View merging ( ``view.merge`` ) to
  optimize the permission checking based on user permissions and layer index
```

  data.
  (`#3736 <https://github.com/vertexproject/synapse/pull/3736>`_)
  (`#3750 <https://github.com/vertexproject/synapse/pull/3750>`_)
  (`#3758 <https://github.com/vertexproject/synapse/pull/3758>`_)

> 

Reviewer: Blank line?

- Add a hotfix that can be used to migrate ``risk:hasvuln`` nodes to
  ``risk:vulnerable`` nodes.
  (`#3745 <https://github.com/vertexproject/synapse/pull/3745>`_)
- Add a Storm API, ``$lib.env.get()``, to get environment variables from
  the Cortex process which start with the prefix ``SYN_STORM_ENV_``.
  (`#3761 <https://github.com/vertexproject/synapse/pull/3761>`_)
- Add a ``edited()`` API to the ``layer`` object in Storm. This API can be
  used to get the last time a given layer was edited. Add a ``reverse``
  argument to the ``layer.edits()`` API to return the node edits in reverse
  order.
  (`#3763 <https://github.com/vertexproject/synapse/pull/3763>`_)
- Add a ``setArchived()`` API to the ``auth:user`` object in Storm.
  (`#3759 <https://github.com/vertexproject/synapse/pull/3759>`_)
- The ``synapse.tool.storm`` tool now returns a non-zero status code when

>   it is invoked to execute a single command and the command encounteres an

Reviewer: ```suggestion
  it is invoked to execute a single command and the command encounters an
```
